sound/soc/sof/intel/hda-stream.c

```diff
@@ -119,13 +119,33 @@ int hda_dsp_stream_setup_bdl(struct snd_sof_dev *sdev,
 	int remain, ioc;
 
 	period_bytes = hstream->period_bytes;
-	dev_dbg(sdev->dev, "period_bytes:0x%x\n", period_bytes);
-	if (!period_bytes)
+	dev_dbg(sdev->dev, "period_bytes: %#x, bufsize: %#x\n", period_bytes,
+		hstream->bufsize);
+
+	/*
+	 * HDA spec demands at least two 'periods' for the DMA to work correctly
+	 * Note: period_bytes == 0 can only happen for firmware or library
+	 * loading, the binary sizes have the necessary alignment requirements
+	 * met
```

Comment:
what alignment requirements are you referring to?

The only requirement for HDaudio DMA is that the start address be aligned on 128 bytes. The size can be arbitrary.

Reply:
Right, I need to move parts of the comment with the new change where I force two periods to get two BLDEs.
The comment tries to say that the firmware size is such that if we split it to two (and it is in a continuous area) then the second chunk will have start address 128-byte align as well.